Submitted By:            Douglas R. Reno <renodr at linuxfromscratch dot org>
Date:                    2018-08-16
Initial Package Version: 1.14.2
Upstream Status:         Submitted
Origin:                  Self
Description:             Fix macro usage in the source code to allow building against
                         glibc-2.28 due to "minor" removal.

diff -Naurp v4l-utils-1.14.2.orig/lib/libv4lconvert/control/libv4lcontrol.c v4l-utils-1.14.2/lib/libv4lconvert/control/libv4lcontrol.c
--- v4l-utils-1.14.2.orig/lib/libv4lconvert/control/libv4lcontrol.c	2017-12-27 08:33:07.000000000 -0600
+++ v4l-utils-1.14.2/lib/libv4lconvert/control/libv4lcontrol.c	2018-08-16 10:28:52.912622258 -0500
@@ -20,9 +20,7 @@
  */
 
 #include <sys/types.h>
-#if defined(MAJOR_IN_SYSMACROS)
 #include <sys/sysmacros.h>
-#endif
 #include <sys/mman.h>
 #include <fcntl.h>
 #include <sys/stat.h>
diff -Naurp v4l-utils-1.14.2.orig/utils/v4l2-ctl/v4l2-ctl.cpp v4l-utils-1.14.2/utils/v4l2-ctl/v4l2-ctl.cpp
--- v4l-utils-1.14.2.orig/utils/v4l2-ctl/v4l2-ctl.cpp	2017-12-27 07:50:55.000000000 -0600
+++ v4l-utils-1.14.2/utils/v4l2-ctl/v4l2-ctl.cpp	2018-08-16 10:29:17.468459524 -0500
@@ -27,6 +27,7 @@
 #include <inttypes.h>
 #include <getopt.h>
 #include <sys/types.h>
+#include <sys/sysmacros.h>
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <ctype.h>
